### DISCUSSION TITLE
Mimir OS  Enabling Tenant Federation: true in mimir store gateway config is breaking the store component.

### DISCUSSION THREAD

**Discussion by prab2014** [grafana/mimir#9989] [Help and support]
### What is the bug?

Trying to achieve federation as described by the official Grafana mimir channels,

Source : -  https://grafana.com/docs/mimir/latest/manage/secure/authentication-and-authorization/

You can federate queries across multiple tenants by using true in -tenant-federation.enabled=true. When you specify tenant IDs, separate them with a pipe (|) character in the X-Scope-OrgID header, as in the example X-Scope-OrgID: tenant-1|tenant-2|tenant-3.

Trying to set the tenant -federation  field to True for mimir store gateway component but receiving the below error 

**"field tenant_federation not found in type storegateway.Config"**

The purpose is to fetch the query results from the multiple tenants . Bigger question , is tenant federation is not part of mimir OS version? only GEM can have federation?

### How to reproduce it?

Deployed latest mimir using helm chart on Kubernetes . To test the tenant federation, currently, setting the  tenant-federation value as true in the config map directly.

store_gateway:
      **#tenant_federation:
        #enabled: true**           
      sharding_ring:
        heartbeat_period: 1m

Once restarted the config map, store gateway pods are crashing along with the ruler components.

### What did you think would happen?

Setting tenant federation should facilitates the metrics from multiple tenants. As per the mimir official documentation , we should be able to federate our queries to multiple tenants. 

store_gateway: <store_gateway>

tenant_federation:
  # If enabled on all services, queries can be federated across multiple
  # tenants. The tenant IDs involved need to be specified separated by a '|'
  # character in the 'X-Scope-OrgID' header.
  # CLI flag: -tenant-federation.enabled
  [enabled: <boolean> | default = false]**

### What was your environment?

Azure Kubernetes

**Comment by 56quarters** [comment]
Tenant federation _is_ part of OSS Mimir. The tenant federation flag isn't part of the store-gateway configuration block. Please check the docs to see where it should be added as part of the Mimir configuration file or as a CLI flag: https://grafana.com/docs/mimir/latest/configure/configuration-parameters/#configuration-parameters
